### PR TITLE
Update dev requirements

### LINKED ISSUE #266: Running tests fail on Python 3.10
Running tests fail on Python 3.10

```
aiven-client on  link-contributing-to-readme [?] via 🐍 v3.10.0 (python-3.10) 
❯ make test
python3 -m flake8 aiven tests
python3 -m pylint aiven tests
************* Module aiven.client.session
aiven/client/session.py:12:39: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
aiven/client/session.py:22:37: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
************* Module aiven.client.cli
aiven/client/cli.py:135:76: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
aiven/client/cli.py:3846:84: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
************* Module aiven.client.argx
aiven/client/argx.py:126:37: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
************* Module aiven.client.connection_info._utils
aiven/client/connection_info/_utils.py:28:14: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
aiven/client/connection_info/_utils.py:29:14: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
aiven/client/connection_info/_utils.py:30:10: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
aiven/client/connection_info/_utils.py:31:10: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
aiven/client/connection_info/_utils.py:32:12: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)
aiven/client/connection_info/_utils.py:34:11: E1136: Value 'Optional' is unsubscriptable (unsubscriptable-object)

-----------------------------------
Your code has been rated at 9.85/10

make: *** [pylint] Error 2
┌───────────────────────────────────────────────────────────────────────────────
(python-3.10) 
```

This is not a question more like just for your information. So not sure if should be an issue or not. Just leaving it here.

### PR DESCRIPTION
**Pull request by rominf**
# About this change: What it does, why it matters

Fix #266.

After introducing more tests, the `format()` function will be replaced
with f-strings.

### CODE REVIEW

**Comment by rominf** [comment]
Added @aiven/team-security since we've got high severity security warnings.

File: requirements.dev.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 flake8
 # Lock mypy to the same version used in downstream build environments
-mypy==0.910
+mypy==0.931
```
Comment (paulwouters) [inline requirements.dev.txt:3]: can this be pinned to "this or higher" ?

File: requirements.dev.txt [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 # Lock pylint to the same version used in downstream build environments
-pylint==2.6.0
+pylint==2.12.2
```
Comment (paulwouters) [inline requirements.dev.txt:5]: same here

**Review by paulwouters — APPROVED** [review]

File: requirements.dev.txt (hunk 1 above)
Comment (paulwouters) [reply requirements.dev.txt:3]: Sure. I meant change from pinning to exact version to "same major and any minor tihs or higher"

**Review by mhoffm-aiven — APPROVED** [review]
lgtm

**Pull request merged**